### PR TITLE
Andi bugs

### PR DESCRIPTION
**Pull request by cameronmarsh**

### CODE REVIEW

File: apps/andi/lib/andi_web/live/edit_live_view.ex [hunk 1]
```diff
@@ -75,7 +75,8 @@ defmodule AndiWeb.EditLiveView do
 
       <%= live_component(@socket, AndiWeb.EditLiveView.DeleteDatasetModal, visibility: @delete_dataset_modal_visibility, id: @dataset_id) %>
 
-      <div phx-hook="showSnackbar">
+      <div id="edit-page-snackbar" phx-hook="showSnackbar">
+        <div style="display: none;"><%= @click_id %></div>
```
Comment (cameronmarsh) [inline apps/andi/lib/andi_web/live/edit_live_view.ex:79]: This click id is just a random uuid that updates any time the save button is pressed or the snackbar should be triggered. This solves the case when the snackbar wouldn't show if nothing in this div was updated. Kind of a hacky solution

File: apps/andi/lib/andi_web/live/edit_live_view/extract_steps/extract_auth_step_form.ex [hunk 2]
```diff
@@ -40,36 +40,36 @@ defmodule AndiWeb.ExtractSteps.ExtractAuthStepForm do
 
             <div class="extract-auth-step-form__destination">
               <%= label(f, :destination, DisplayNames.get(:destination),  class: "label label--required") %>
-              <%= text_input(f, :destination, class: "input") %>
+              <%= text_input(f, :destination, id: "step_#{@id}__auth_destination", class: "input") %>
```
Comment (cameronmarsh) [inline apps/andi/lib/andi_web/live/edit_live_view/extract_steps/extract_auth_step_form.ex:43]: apparently having duplicate ids on liveview elements caused the issue where new steps of the same type would blank out the fields of old steps

File: apps/andi/lib/andi_web/live/edit_live_view/extract_steps/extract_http_step_form.ex [hunk 3]
```diff
@@ -137,7 +137,15 @@ defmodule AndiWeb.ExtractSteps.ExtractHttpStepForm do
       |> Ecto.Changeset.get_field(:queryParams)
       |> remove_key_value(query_param_id)
 
-    new_changset = Ecto.Changeset.put_embed(socket.assigns.changeset, :queryParams, updated_query_params)
+    updated_url =
```
Comment (cameronmarsh) [inline apps/andi/lib/andi_web/live/edit_live_view/extract_steps/extract_http_step_form.ex:140]: the url wasn't being updated when query params were being removed

File: apps/reaper/lib/reaper/data_extract/extract_step.ex [hunk 4]
```diff
@@ -11,6 +11,7 @@ defmodule Reaper.DataExtract.ExtractStep do
 
   def execute_extract_steps(dataset, steps) do
     Enum.reduce(steps, %{}, fn step, acc ->
+      step = AtomicMap.convert(step, underscore: false)
```
Comment (cameronmarsh) [inline apps/reaper/lib/reaper/data_extract/extract_step.ex:14]: there were issues around extract steps having string keys sometimes

File: apps/reaper/lib/reaper/url_builder.ex [hunk 5]
```diff
@@ -32,12 +32,16 @@ defmodule Reaper.UrlBuilder do
   end
 
   def decode_http_extract_step(%{context: %{url: url, queryParams: query_params}, assigns: assigns}) do
-    string_params =
-      query_params
-      |> safe_evaluate_parameters(assigns)
-      |> URI.encode_query()
-
-    "#{build_safe_url_path(url, assigns)}?#{string_params}"
+      case url_has_query_params?(url) do
```
Comment (cameronmarsh) [inline apps/reaper/lib/reaper/url_builder.ex:35]: fixes the query params being added twice to an extract url

**Review by kcarra — APPROVED** [review]

**Pull request merged**